torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -546,16 +662,21 @@ def fuse_all_gather_matmul(all_gather: _AllGatherMatch) -> None:
         return
 
     # Find consumer matmuls
-    matmuls = _find_consumer_matmuls(ag_res_node)
+    if with_effects:
+        with_effects_output_token, with_effects_result = with_effects_users(
+            with_effects
+        )
+        if ag_res_node == with_effects:
+            ag_res_node = with_effects_result
```

Comment:
This logic should be in `_find_consumer_matmuls`. We want to isolate `all_gather` pattern matching and the operator fusion, so that future changes to the all-gather graph patterns can be isolated.